src/pages/settings/ProfilePage.js

```diff
@@ -1,24 +1,236 @@
+        avatarURL: PropTypes.string,
+
+        // Pronouns of the current user from their personal details
+        pronouns: PropTypes.string,
+
+        // timezone of the current user from their personal details
+        timezone: PropTypes.shape({
+
+            // Value of selected timezone
+            selected: PropTypes.string,
+
+            // Whether timezone is automatically set
+            automatic: PropTypes.bool,
+        }),
+    }),
+};
+
+const defaultProps = {
+    myPersonalDetails: {},
+};
+
+const pronounsMap = {
+    theyThemTheirs: 'They/them/theirs',
+    sheHerHers: 'She/her/hers',
+    heHimHis: 'He/him/his',
+    zeHirHirs: 'Ze/hir/hirs',
+    selfSelect: 'Self-select',
+    callMeByMyName: 'Call me by my name',
+};
+
+const timezones = moment.tz.names()
+    .map(timezone => ({
+        value: timezone,
+        label: timezone,
+    }));
+
+class ProfilePage extends Component {
+    constructor(props) {
+        super(props);
+
+        const {
+            firstName, lastName, pronouns, timezone = {},
+        } = props.myPersonalDetails;
+
+        this.state = {
+            firstName,
+            lastName,
+            pronouns,
+            selfSelectedPronouns: '',
+            selectedTimezone: timezone.selected ?? CONST.DEFAULT_TIME_ZONE.selected,
+            isAutomaticTimezone: timezone.automatic ?? CONST.DEFAULT_TIME_ZONE.automatic,
+        };
+
+        this.pronounDropdownValues = Object.values(pronounsMap).map(pronoun => ({value: pronoun, label: pronoun}));
+        this.updatePersonalDetails = this.updatePersonalDetails.bind(this);
+    }
+
+    updatePersonalDetails() {
+        const {
+            firstName, lastName, pronouns, selfSelectedPronouns, selectedTimezone, isAutomaticTimezone,
+        } = this.state;
+
+        setPersonalDetails({
+            firstName,
+            lastName,
+            pronouns: pronouns === pronounsMap.selfSelect ? selfSelectedPronouns : pronouns,
+            timezone: {
+                automatic: isAutomaticTimezone,
+                selected: selectedTimezone,
+            },
+        });
+    }
+
+    render() {
+        return (
+            <ScreenWrapper>
+                {() => (
+                    <>
+                        <HeaderWithCloseButton
+                            title="Profile"
+                            shouldShowBackButton
+                            onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS)}
+                            onCloseButtonPress={Navigation.dismissModal}
+                        />
+                        <View style={styles.p5}>
+                            <Avatar
+                                style={[styles.avatarLarge, styles.alignSelfCenter]}
+                                source={this.props.myPersonalDetails.avatarURL}
+                            />
+                            <Text fontSize={17} style={[styles.mt6, styles.mb6]}>
```

Comment:
Left a similar comment in your other PR - but let's just use our standard `styles.textP` here.